src/v/storage/disk_log_impl.cc

```diff
@@ -559,9 +566,7 @@ ss::future<bool> disk_log_impl::sliding_window_compact(
         if (cfg.asrc) {
             cfg.asrc->check();
         }
-        if (seg->finished_self_compaction()) {
-            continue;
-        }
+
```

Comment:
The removed code here was a superfluous check that we perform in `self_compact_segment()` anyways. move `continue` logic after call by checking for `result.did_compact() == false`.